crates/sp-domains/src/lib.rs

```diff
@@ -515,6 +516,11 @@ impl DomainsDigestItem for DigestItem {
 pub struct DomainInstanceData {
     pub runtime_type: RuntimeType,
     pub runtime_code: Vec<u8>,
+    // The genesis config of the domain, encoded in json format.
+    //
+    // NOTE: the WASM code in the `system-pallet` genesis config should be empty to avoid
+    // redundancy with the `runtime_code` field.
+    pub raw_genesis_config: Option<Vec<u8>>,
```

Comment:
Why is this required here ? We want the Genesiraw genesis to only patch the Domain genesis when the chain is in Dev or Local test mode. For all other scenarios, we never should use it.

Reply:
https://github.com/subspace/subspace/pull/1704#discussion_r1277422781 may answer this, `DomainInstanceData::raw_genesis_config` is only set to `Some` for the genesis domain, and set to `None` for other domain instantiated by the `instantiate_domain` call.